File: backend/aihub-api/src/main/java/com/aihub/service/SetupService.java
package com.aihub.service;

import com.aihub.dto.ConnectionTestResult;
import com.aihub.dto.DatabaseConfigDTO;

public interface SetupService {
    
    /**
     * 测试数据库连接
     * @param config 数据库配置
     * @return 连接测试结果
     */
    ConnectionTestResult testConnection(DatabaseConfigDTO config);
    
    /**
     * 保存数据库配置到本地文件
     * @param config 数据库配置
     */
    void saveConfig(DatabaseConfigDTO config);
    
    /**
     * 检查是否已配置数据库
     * @return true-已配置，false-未配置
     */
    boolean isConfigured();
}

File: frontend/src/services/setupService.ts
/**
 * 配置服务
 */
import apiClient from './apiClient'
import type { DatabaseConfig, ConnectionTestResult } from '../types/api'

export const setupService = {
  /**
   * 检查是否已配置数据库
   */
  checkSetupStatus: async (): Promise<boolean> => {
    return await apiClient.get<boolean>('/setup/status')
  },

  /**
   * 测试数据库连接
   */
  testConnection: async (config: DatabaseConfig): Promise<ConnectionTestResult> => {
    return await apiClient.post<ConnectionTestResult>('/setup/test-connection', config)
  },

  /**
   * 保存数据库配置
   */
  saveConfig: async (config: DatabaseConfig): Promise<void> => {
    await apiClient.post('/setup/save-config', config)
  },
}
